### ISSUE TITLE
com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_twoStage

### ISSUE THREAD

**Issue by cangencer** [hazelcast/hazelcast-jet#899]
https://hazelcast-l337.ci.cloudbees.com/view/Jet/job/jet-jdk9/com.hazelcast.jet$hazelcast-jet-core/209/testReport/junit/com.hazelcast.jet.core/JobRestartWithSnapshotTest/when_nodeDown_then_jobRestartsFromSnapshot_twoStage/

```Error Message
No snapshot record found - job likely finished
Stacktrace
java.lang.AssertionError: No snapshot record found - job likely finished
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.lambda$waitForNextSnapshot$9(JobRestartWithSnapshotTest.java:364)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.waitForNextSnapshot(JobRestartWithSnapshotTest.java:362)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot(JobRestartWithSnapshotTest.java:218)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_twoStage(JobRestartWithSnapshotTest.java:117)```

**Comment by cangencer** [comment]
possibly related.. 


com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_singleStage

```
Error Message
No snapshot record found - job likely finished
Stacktrace
java.lang.AssertionError: No snapshot record found - job likely finished
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.lambda$waitForNextSnapshot$9(JobRestartWithSnapshotTest.java:364)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.waitForNextSnapshot(JobRestartWithSnapshotTest.java:362)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot(JobRestartWithSnapshotTest.java:218)
	at com.hazelcast.jet.core.JobRestartWithSnapshotTest.when_nodeDown_then_jobRestartsFromSnapshot_singleStage(JobRestartWithSnapshotTest.java:112)
```

https://hazelcast-l337.ci.cloudbees.com/view/Jet/job/jet-sonar/com.hazelcast.jet$hazelcast-jet-core/1261/testReport/junit/com.hazelcast.jet.core/JobRestartWithSnapshotTest/when_nodeDown_then_jobRestartsFromSnapshot_singleStage/

**Comment by viliam-durina** [comment]
Fixed in #907.
